### PR TITLE
2732: Add language change for local news

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Add possibility to directly change the language for local news.

### Proposed changes

- Use the news id as number (instead of stringified)
- Add `availableLanguages` to `LocalNewsModel`
- Implement language change for web and native

### Side effects

None.

### Resolved issues

Fixes: #2732.

---

### CODE REVIEW

File: release-notes/unreleased/2732-news-language-change.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+issue_key: 2732
+show_in_stores: true
+platforms:
+  - android
+  - ios
+  - web
+en: Enable language change for local news.
+de: Sprachwechsel für Lokale Veranstaltungen wird nun unterstützt.
```
Comment (LeandraH) [inline release-notes/unreleased/2732-news-language-change.yml:8]:
```suggestion
de: Sprachwechsel für lokale Nachrichten wird nun unterstützt.
```

File: web/src/routes/LocalNewsPage.tsx [hunk 2]
```diff
@@ -43,27 +59,22 @@ const LocalNewsPage = ({ city, pathname, languageCode, cityCode }: CityRouteProp
...
-          newsId: id.toString(),
-        })}
+        link={createNewsPath({ newsId: id })}
         t={t}
         type={LOCAL_NEWS_TYPE}
       />
     )
   }
 
   // Language change is not possible between local news detail views because we don't know the id of other languages
```
Comment (LeandraH) [inline web/src/routes/LocalNewsPage.tsx:69]: I think the comment can be removed, or updated :)

**Review by LeandraH — COMMENTED** [review]
Thanks for the PR! It's nice to see you're back 🤗 

A few things though:

1. I'd love some tests in `shared` for actually getting some news ids, mostly as a kind of documentation, but that's optional.

2. When I go to http://localhost:9000/testumgebung/de/news/local/7154 and then set the language to English, I am sent to a new page but that one just says that the page doesn't exist in the selected language, and asks if I want to return to the German one. That seems like a pretty bad user experience to me. What do you think? (I know, it's a CMS issue anyway)

3. In iOS, when I go to the same news story in English, I get the same message that it's only available in German; but when I try to switch to German in the app (not by following the link), I get an error that no translation is available.
simulator_screenshot_94E8E5A6-02EE-4CCC-91BD-5667E183259D

Oh, in fact, I can't switch language in iOS or Android at all when in a news story :(

**Comment by LeandraH** [comment]
Uhm, this is weird, I can't change the language in emulated iOS or emulated Android, but on my real iPhone, it does work.

**Comment by steffenkleinle** [comment]
> Thanks for the PR! It's nice to see you're back 🤗

:heart: 

>     1. I'd love some tests in `shared` for actually getting some news ids, mostly as a kind of documentation, but that's optional.

Sorry, I don't understand this. What would you like tests for?


> 
>     2. When I go to http://localhost:9000/testumgebung/de/news/local/7154 and then set the language to English, I am sent to a new page but that one just says that the page doesn't exist in the selected language, and asks if I want to return to the German one. That seems like a pretty bad user experience to me. What do you think? (I know, it's a CMS issue anyway)
> 
>     3. In iOS, when I go to the same news story in English, I get the same message that it's only available in German; but when I try to switch to German in the app (not by following the link), I get an error that no translation is available.
> 
> Oh, in fact, I can't switch language in iOS or Android at all when in a news story :(

**Comment by steffenkleinle** [comment]
> Uhm, this is weird, I can't change the language in emulated iOS or emulated Android, but on my real iPhone, it does work. But there, only for the local ones, tünews also won't let me change the language.

TüNews is not done in this PR, we don't have information about translations.
Hmm, you are sure you are on the right branch everywhere? Please also include screenrecords or something, I am not sure what this means (i.e. what does `I can't change the language` means. Does the language change button not work? Are there no available languages displayed? Is an error displayed after a language is selected?)

**Comment by LeandraH** [comment]
> > Uhm, this is weird, I can't change the language in emulated iOS or emulated Android, but on my real iPhone, it does work. But there, only for the local ones, tünews also won't let me change the language.
> 
> TüNews is not done in this PR, we don't have information about translations. Hmm, you are sure you are on the right branch everywhere? Please also include screenrecords or something, I am not sure what this means (i.e. what does `I can't change the language` means. Does the language change button not work? Are there no available languages displayed? Is an error displayed after a language is selected?)

Sorry, what happens when I can't change the language is that I tap on the icon that opens the overlay to change the language, and it doesn't open and I instead get the snackbar with the error in the screenshot, "No translation available". I'll test again to make sure I am on the right branch :)

**Comment by LeandraH** [comment]
https://github.com/user-attachments/assets/76f495d3-8019-4fe2-bac0-25deef7d1eef

**Comment by steffenkleinle** [comment]
> > > Uhm, this is weird, I can't change the language in emulated iOS or emulated Android, but on my real iPhone, it does work. But there, only for the local ones, tünews also won't let me change the language.
> > 
> > 
> > TüNews is not done in this PR, we don't have information about translations. Hmm, you are sure you are on the right branch everywhere? Please also include screenrecords or something, I am not sure what this means (i.e. what does `I can't change the language` means. Does the language change button not work? Are there no available languages displayed? Is an error displayed after a language is selected?)
> 
> Sorry, what happens when I can't change the language is that I tap on the icon that opens the overlay to change the language, and it doesn't open and I instead get the snackbar with the error in the screenshot, "No translation available". I'll test again to make sure I am on the right branch :)

Hmmm, that is so weird... I have no idea what the problem is here... Does it work on web for you? Could you perhaps log the available languages property and try to debug that? For me it works as expected, no idea what is causing this for you. :/


https://github.com/user-attachments/assets/a9912a09-79c2-4088-9631-39d3239f8825

**Comment by LeandraH** [comment]
> Hmmm, that is so weird... I have no idea what the problem is here... Does it work on web for you? Could you perhaps log the available languages property and try to debug that? For me it works as expected, no idea what is causing this for you. :/

It works on web for me, I'll try debugging a bit 🤞

**Review by LeandraH — APPROVED** [review]
Grrrrr, hot reloading was broken again. Sorry about that. It works on emulated and real iOS. And in Firefox.

**Review by f1sh1918 — APPROVED** [review]
Works fine. Tested on android and chrome 👍

**Pull request merged**